<launch>
  <arg name="gui" default="true"/>

  <include file="$(find jsk_pcl_ros_utils)/sample/sample_depth_image_error.launch">
    <arg name="gui" value="false"/>
  </include>

  <node name="depth_calibration"
        pkg="nodelet" type="nodelet"
        args="standalone jsk_pcl/DepthCalibration">
    <remap from="~input" to="camera/depth_registered/sw_registered/image_rect"/>
    <remap from="~camera_info" to="camera/depth_registered/sw_registered/camera_info"/>
  </node>

  <node name="depth_error_calibration"
        pkg="jsk_pcl_ros" type="depth_error_calibration.py"
        output="screen">
    <remap from="/camera_remote/depth_calibration/set_calibration_parameter"
           to="depth_calibration/set_calibration_parameter"/>
    <rosparam>
      dump_result_into_yaml: false
    </rosparam>
  </node>

  <group if="$(arg gui)">
    <node name="input_depth_viz"
          pkg="image_view" type="image_view">
      <remap from="image" to="camera/depth_registered/sw_registered/image_rect"/>
      <rosparam>
        gui: false
        min_image_value: 0.8
        max_image_value: 2.0
      </rosparam>
    </node>

    <node name="output_depth_viz"
          pkg="image_view" type="image_view">
      <remap from="image" to="depth_calibration/output"/>
      <rosparam>
        gui: false
        min_image_value: 0.8
        max_image_value: 2.0
      </rosparam>
    </node>

    <node name="tile_image"
          pkg="jsk_perception" type="tile_image.py">
      <rosparam>
        input_topics:
          - /input_depth_viz/output
          - /output_depth_viz/output
          - /depth_error_calibration/error_plot_image
        no_sync: true
        draw_topic_name: true
        font_scale: 0.9
      </rosparam>
    </node>

    <node name="image_view"
          pkg="image_view" type="image_view">
      <remap from="image" to="tile_image/output"/>
    </node>
  </group>
</launch>
